ProcessLib/RichardsFlow/RichardsFlowFEM.h

```diff
@@ -151,63 +155,87 @@ class LocalAssemblerData : public RichardsFlowLocalAssemblerInterface
             _integration_method.getNumberOfPoints();
         ParameterLib::SpatialPosition pos;
         pos.setElementID(_element.getID());
-        const int material_id =
-            _process_data.material->getMaterialID(_element.getID());
-        const Eigen::MatrixXd& perm = _process_data.material->getPermeability(
-            material_id, t, pos, _element.getDimension());
-        assert(perm.rows() == _element.getDimension() || perm.rows() == 1);
-        GlobalDimMatrixType permeability = GlobalDimMatrixType::Zero(
-            _element.getDimension(), _element.getDimension());
-        if (perm.rows() == _element.getDimension())
-        {
-            permeability = perm;
-        }
-        else if (perm.rows() == 1)
-        {
-            permeability.diagonal().setConstant(perm(0, 0));
-        }
+
+        auto const& medium =
+            *_process_data.media_map->getMedium(_element.getID());
+        auto const& liquid_phase = medium.phase("AqueousLiquid");
+        auto const& solid_phase = medium.phase("Solid");
+        MaterialPropertyLib::VariableArray vars;
+        vars[static_cast<int>(MaterialPropertyLib::Variable::temperature)] =
+            medium
+                .property(
+                    MaterialPropertyLib::PropertyType::reference_temperature)
+                .template value<double>(vars, pos, t, dt);
 
         for (unsigned ip = 0; ip < n_integration_points; ip++)
         {
             pos.setIntegrationPoint(ip);
             double p_int_pt = 0.0;
             NumLib::shapeFunctionInterpolate(local_x, _ip_data[ip].N, p_int_pt);
-            const double& temperature = _process_data.temperature(t, pos)[0];
-            auto const porosity = _process_data.material->getPorosity(
-                material_id, t, pos, p_int_pt, temperature, 0);
 
-            double const pc_int_pt = -p_int_pt;
+            vars[static_cast<int>(
+                MaterialPropertyLib::Variable::phase_pressure)] = p_int_pt;
+
+            auto const permeability =
+                MaterialPropertyLib::formEigenTensor<GlobalDim>(
+                    solid_phase.property(MaterialPropertyLib::permeability)
+                        .value(vars, pos, t, dt));
+
+            auto const porosity =
+                solid_phase.property(MaterialPropertyLib::PropertyType::porosity)
+                    .template value<double>(vars, pos, t, dt);
 
-            double const Sw = _process_data.material->getSaturation(
-                material_id, t, pos, p_int_pt, temperature, pc_int_pt);
+            vars[static_cast<int>(
+                MaterialPropertyLib::Variable::capillary_pressure)] = -p_int_pt;
```

Comment:
Please move these two lines to the line after line 177, where `MaterialPropertyLib::Variable::phase_pressure` is set.
```suggestion
            vars[static_cast<int>(
                MaterialPropertyLib::Variable::capillary_pressure)] = -p_int_pt;
```